src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,139 @@
+time**
+
+`<cdk-virtual-scroll-viewport>` displays large lists of elements performantly by only
+rendering the items that fit on-screen. This components works with `*cdkVirtualFor`,
+a virtualized replacement for `*ngFor`.
+
+<!-- TODO(mmalerba): Add embedded examples once this is available in @angular/cdk -->
+For some example usages, 
+[see the demo app](https://github.com/angular/material2/tree/master/src/demo-app/virtual-scroll).
+
+### Creating items in the viewport
+`*cdkVirtualFor` replaces `*ngFor` inside of a `<cdk-virtual-scroll-viewport>`, supporting the exact
+same API as [`*ngFor`](https://angular.io/api/common/NgForOf).
+The simplest usage just specifies the list of items:
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+`*cdkVirtualFor` makes the following context variables available to the template:
+
+| Context variable | Description                                        |
+|------------------|----------------------------------------------------|
+| `index`          | The index of the item in the data source.          |
+| `count`          | The total number of items in the data source.      |
+| `first`          | Whether this is the first item in the data source. |
+| `last`           | Whether this is the last item in the data source.  |
+| `even`           | Whether the `index` is even.                       |
+| `odd`            | Whether the `index` is odd.                        |
+
+All of these apply to the index of the item in the data source, not the index in the rendered
+portion of the data.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items;
+                       let index = index;
+                       let count = count">
+    {{item}} ({{index}} of {{count}})
+  </div>
+</cdk-virtual-scroll-viewport>
+```
+
+A `trackBy` function can be specified and works the same as the `*ngFor` `trackBy`. The `index`
+passed to the tracking function will be the index in the data source, not the index in the rendered
+portion.
+
+#### View recycling
+To improve rendering performance, `*cdkVirtualFor` caches previously created views after
+they are no longer needed. When a new view would normally be created, a cached view
+is reused instead. The size of the view cache can be adjusted via the `templateCacheSize`
+property; setting this size to `0` disables caching. If your templates are expensive in terms of
+memory you may wish to reduce this number to avoid spending too much memory on the template cache.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items; templateCacheSize: 0">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+#### Specifying data
+`*cdkVirtualFor` accepts data from an `Array`, `Observable<Array>`, or `DataSource`. The
```

Comment:
I'd show an example code snippet here.

Reply:
I'll add that once we move it into the cdk proper and have actual embedded examples